frida_tools/tracer.py

```diff
@@ -54,6 +54,8 @@ def process_builder_arg(option, opt_str, value, parser, method, **kwargs):
                     type='string', action='callback', callback=process_builder_arg, callback_args=(pb.include_debug_symbol,))
             parser.add_option("-q", "--quiet", help="do not format output messages", action='store_true', default=False)
             parser.add_option("-d", "--decorate", help="Add module name to generated onEnter log statement", action='store_true', default=False)
+            parser.add_option("-k", "--init-session", help="Path to Javascript file used to initialize the session", metavar="PATH",
+                    type='string', action='callback', callback=process_builder_arg, callback_args=(pb.init_session_paths,))
```

Comment:
Does this belong on the builder though? It looks like the logic is outside it, so we probably shouldn't store the state on it.

Reply:
I wanted to piggyback on the same logic used by the other command line options: store the --init-session paths during parsing, then process them when generating the JS trace code.  Can you suggest a better home for handling --init-session options?

Reply:
I would store it on `self` like the other regular options.

Reply:
There can be multiple `init-session` options on the command line, so I need the callback method for storing all of the script paths.  Should `--init-session` be handled any differently than, say, `--include`?

Reply:
Well if it integrates with the builder then it should be stored there, but it doesn't so that's why I was thinking it should be stored outside. So if we store it outside, we should be able to use the `append` action.

Reply:
It's not clear to me why some options (the regulars) go into `self` while others go into `builder`.  What is the concept of builder vs. non-builder?

Nonetheless, I have made the change, which is now much cleaner.  No need for callbacks to builder, the multiple -S parameters are appended to `self`.  Good point.

Reply:
Awesome, thanks!

The builder is currently only meant to construct the list of includes/excludes that specifies what to trace. But in retrospect, given its name Tracer**Profile**Builder, that name does suggest that it's supposed to specify a lot more than just the *what*. So it would make sense to rename it. 🤔 Ah well, let's keep that in mind for later.